fearless/Common/Services/GitHubPhishingService/GitHubPhishingAPIService.swift

```diff
@@ -44,26 +51,24 @@ class GitHubPhishingAPIService: ApplicationServiceProtocol {
     }
 
     private func clearConnection() {
-        operation.cancel()
+        networkOperation.cancel()
     }
 
     private func setupConnection() {
-        operation.completionBlock = { [self] in
-            do {
-                if let phishingItems = try operation.extractResultData() {
-                    let deleteOperation = storage.deleteAllOperation()
-                    OperationManagerFacade.sharedManager.enqueue(operations: [deleteOperation], in: .sync)
-
-                    for phishingItem in phishingItems {
-                        let saveOperation = storage.saveOperation({ [phishingItem] }, { [] })
-                        OperationManagerFacade.sharedManager.enqueue(operations: [saveOperation], in: .sync)
-                    }
-                }
-            } catch {
-                self.logger.error("Request unsuccessful")
-            }
+
+        networkOperation = GitHubOperationFactory().fetchPhishingListOperation(url)
+
+        let replaceOperation = storage.replaceOperation {
+            let phishingItem = try self.networkOperation
+                .extractResultData(throwing: BaseOperationError.parentOperationCancelled)
+            return phishingItem
         }
 
-        OperationManagerFacade.sharedManager.enqueue(operations: [operation], in: .sync)
+        replaceOperation.addDependency(networkOperation)
+
+        let operationWrapper = CompoundOperationWrapper(targetOperation: replaceOperation,
+                                        dependencies: [networkOperation])
+
+        OperationManagerFacade.sharedManager.enqueue(operations: operationWrapper.allOperations, in: .sync)
```

Comment:
We have passed operation manager but don't use it

Reply:
Opps. Fixed